### DISCUSSION TITLE
Read only SQLite with prisma on Vercel

### DISCUSSION THREAD

**Discussion by ethernal** [keystonejs/keystone#8990] [Questions]
Hello,

I have seen the example of application running sqlite in read-only mode deployed to Vercel. From what I've seen it's rewriting the path to the db when instantiating PrismaClient.. but when I tried this in pure Next.js app it did not work. Does anyone have any experience with it or could elaborate on how it was done in Keystone? I am trying to replicate the functionality as it seems way faster than file based (MDX) structure for data but I am not ready to rewrite it in Keystone.. yet.

Thank you.

Here's the link to working SQLite example of my site but it's without a DB access: https://github.com/ethernal/personal-site/tree/posts-to-db

I had a separate branch when I needed to commit the DB but it would not work either so I am curious what is it that I am missing. Also having it read only is quite fine as this is only a performance and feature unlock (can query for related data etc). 

Thank you!

**Comment by johnnysn** [comment]
Hi there,

I managed to access the sqlite database file in production by getting the current working directory before configuring the Prisma client. Here's how:

```typescript
import { PrismaClient } from '@prisma/client';
import path from 'path';

declare global {
  // eslint-disable-next-line no-var
  var cachedPrisma: PrismaClient;
}

// Workaround to find the db file in production
const filePath = path.join(process.cwd(), 'prisma/local.db');
const config = {
  datasources: {
    db: {
      url: 'file:' + filePath,
    },
  },
};

let prisma: PrismaClient;
if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient(config);
} else {
  if (!global.cachedPrisma) {
    global.cachedPrisma = new PrismaClient(config);
  }
  prisma = global.cachedPrisma;
}

export const db = prisma;
```

Let me know how it worked out for you ;). 
Cheers!

**Reply by cademis** [reply]
This worked perfectly for me thanks! A great solution for demo projects

**Reply by Hamzashahzad26** [reply]
how to do this kindly guide me

**Comment by Hamzashahzad26** [comment]
where we have to copy this code

**Reply by JoKenPo** [reply]
on the prisma client file, like this [example on next ](https://github.com/prisma/prisma-examples/blob/latest/typescript/rest-nextjs-api-routes-auth/lib/prisma.ts)
